### PR TITLE
Add crop image filter

### PR DESCRIPTION
**Pull request by timheap**
It behaves a bit like the CSS property `background-size: contain`, but is actually most useful with `background-size: cover` for full-width, fixed-height banner images in page headers.

### CODE REVIEW

**Comment by davecranwell** [comment]
Does this go any way towards addressing https://github.com/torchbox/wagtail/issues/595 ? 

From what I can see it still doesn't upscale so would return images which don't match the dimensions requested, or am I mistaken?

File: wagtail/wagtailimages/tests/test_image_operations.py [hunk 1]
```diff
@@ -325,6 +325,68 @@ class TestWidthHeightOperation(ImageOperationTestCase):
...
+        # Both dimensions set, image larger than both
+        ('crop-200x200', Image(width=600, height=400), [
+            ('resize', ((300, 200), ), {}),
+            ('crop', ((50, 0, 250, 200), ), {}),
+        ]),
+        ('crop-200x200', Image(width=400, height=600), [
+            ('resize', ((200, 300), ), {}),
+            ('crop', ((0, 50, 200, 250), ), {}),
+        ]),
+        ('crop-200x200', Image(width=400, height=400), [
```
Comment (kaedroho) [inline wagtail/wagtailimages/tests/test_image_operations.py]: I got a feeling that each run test needs to have a different spec or the method names will clash. We should fix that.

File: docs/images/using_in_templates.rst [hunk 2]
```diff
@@ -104,6 +104,27 @@ The available resizing methods are:
...
+
+            {% image self.banner crop-100x200 %}
+
+        Crop the image so it fits within the given dimensions.
+
+        Both dimensions are optional. If they are left blank, that dimension is not cropped. This is useful for cropping banner images for page headers to a specific height, but leaving the image as wide as possible:
+
+        .. code-block:: django
+
+            {% image self.banner crop-x150 %}}
```
Comment (zerolab) [inline docs/images/using_in_templates.rst]: Typo: `%}}`

File: wagtail/wagtailimages/tests/test_image_operations.py (hunk 1 above)
Comment (timheap) [inline wagtail/wagtailimages/tests/test_image_operations.py]: You're right.  I had actually fixed this as well, but I realise now that I  accidentally revoked that change when I was preparing this commit!

**Comment by timheap** [comment]
@davecranwell You're right that it does not upscale images, and does not try to preserve aspect ratio. I updated the docs, and how it works is explained in detail there, but essentially I made it for the page banner images on <http://digitalready.takeflight.net.au/> (WIP site, hence the lack of content, but the various pages have banner images). I needed something that would produce images that retained their natural width, but were cropped to 125 pixels high.

**Comment by timheap** [comment]
I've stopped the generated image operation test names from colliding.

**Comment by kaedroho** [comment]
Thanks @timheap!

This is very similar to the ``fill`` operation that we already have and in many cases it gives the same result.

Could we change the ``fill`` filter instead to allow the developer to leave off either the width or height?

**Comment by timheap** [comment]
Because you can add image operations on a per project basis, I'll just continue adding this to projects as I need it. No one except me seems interested in this operation.

**Pull request closed without merging**